scylla/src/transport/iterator.rs

```diff
@@ -683,6 +715,66 @@ where
     }
 }
 
+/// A massively simplified version of the RowIteratorWorker. It does not have
+/// any complicated logic related to retries, it just fetches pages from
+/// a single connection.
```

Comment:
It would be nice to use a `RetryPolicy` for retries. I think it wouldn't be too hard to incorporate it.

Reply:
I'm not sure whether it makes sense for queries that are limited to a single connection and only read data from a single node. For example: what should happen on `RetryNextNode`?